CRM/Financial/BAO/Payment.php

```diff
@@ -352,4 +346,141 @@ public static function recordRefundPayment($contributionId, $trxnData, $updateSt
     return $financialTrxn;
   }
 
+  /**
+   * @param int $contributionId
+   * @param array $trxnData
+   * @param int $participantId
+   *
+   * @return \CRM_Core_BAO_FinancialTrxn
+   */
+  public static function recordPayment($contributionId, $trxnData, $participantId) {
```

Comment:
Agree with this name as we will be using this for recording payment for both pending and partially paid contribution, otherwise, I would have suggested recordPartialPayment()

Reply:
I think we need to ditch the partial payment wording as a whole - payments are not partial (although the contribution may or may not be)